internal/impl/generic/cork_test.go

```diff
@@ -0,0 +1,247 @@
+package generic
+
+import (
+	"context"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func createSignalSocket(t *testing.T) string {
+	tmpDir, err := os.MkdirTemp("", "benthos_cork_test")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		require.NoError(t, os.RemoveAll(tmpDir))
+	})
+
+	return filepath.Join(tmpDir, "signal.sock")
+}
+
+func testCork(t *testing.T, cork *corkInput, conn net.Conn) {
+	// send a message to uncork
+	bs := []byte("cork\n")
+	n, err := conn.Write(bs)
+	require.NoError(t, err)
+	require.Equal(t, len(bs), n)
+	require.Eventually(t, func() bool {
+		return cork.Corked()
+	}, 1*time.Second, 100*time.Millisecond, "input was not corked in time")
+}
+
+func testUncork(t *testing.T, cork *corkInput, conn net.Conn) {
+	// send a message to uncork
+	bs := []byte("uncork\n")
+	n, err := conn.Write(bs)
+	require.NoError(t, err)
+	require.Equal(t, len(bs), n)
+	require.Eventually(t, func() bool {
+		return !cork.Corked()
+	}, 1*time.Second, 100*time.Millisecond, "input was not uncorked in time")
+}
+
+func testNilRead(t *testing.T, cork *corkInput) {
+	ctx := context.Background()
+
+	batch, ack, err := cork.ReadBatch(ctx)
+	require.NoError(t, err)
+	require.Nil(t, batch)
+	err = ack(ctx, nil)
+	require.NoError(t, err)
+}
+
+// read a batch from the input and test its contents
+// we assume test inputs yield batches with a single message
+func testRealRead(t *testing.T, cork *corkInput, expectedValue string) {
+	ctx := context.Background()
+
+	batch, ack, err := cork.ReadBatch(ctx)
+	require.NoError(t, err)
+	require.Len(t, batch, 1)
+	err = ack(ctx, nil)
+	require.NoError(t, err)
+
+	// assert message contents
+	msg := batch[0]
+	bs, err := msg.AsBytes()
+	require.NoError(t, err)
+	require.Equal(t, string(bs), expectedValue)
+}
+
+func TestCork(t *testing.T) {
+	socketFile := createSignalSocket(t)
+	ctx := context.Background()
+	spec := newCorkConfigSpec()
+
+	cfg, err := spec.ParseYAML(`
+signal:
+  socket_server:
+    network: unix
+    address: `+socketFile+`
+    codec: lines
```

Comment:
I couldn't think of a way to pass a mock input that would let me pass signals in my test code so i opted for this approach.